packages/components/src/components/hds/form/fieldset/index.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { getElementId } from '../../../../utils/hds-get-element-id.ts';
+import {
+  ariaDescribedBy,
+  registerAriaDescriptionElement,
+  unregisterAriaDescriptionElement,
+} from '../../../../utils/hds-aria-described-by.ts';
+import HdsFormLegendComponent from '../legend/index.ts';
+import HdsFormHelperTextComponent from '../helper-text/index.ts';
+import HdsFormErrorComponent from '../error/index.ts';
+
+import type { HdsFormLayouts } from '../types.ts';
+import type { ComponentLike, WithBoundArgs } from '@glint/template';
+import type { HdsYieldSignature } from '../../yield/index.ts';
+
+export interface HdsFormFieldsetSignature {
+  Args: {
+    isOptional?: boolean;
+    isRequired?: boolean;
+    layout?: HdsFormLayouts;
+  };
+  Blocks: {
+    default: [
+      {
+        Legend?: WithBoundArgs<
+          typeof HdsFormLegendComponent,
+          'contextualClass' | 'isRequired' | 'isOptional'
+        >;
+        HelperText?: WithBoundArgs<
+          typeof HdsFormHelperTextComponent,
+          'contextualClass' | 'controlId' | 'onInsert'
+        >;
+        Control?: ComponentLike<HdsYieldSignature>;
+        Error?: WithBoundArgs<
+          typeof HdsFormErrorComponent,
+          'contextualClass' | 'controlId' | 'onInsert' | 'onRemove'
+        >;
+        id?: string;
+        ariaDescribedBy?: string;
+      },
+    ];
+  };
+  Element: HTMLElement;
+}
+
+// @ts-expect-error: decorator function return type 'ClassOf<AriaDescribedByComponent>' is not assignable to 'typeof HdsFormFieldComponent'
+@ariaDescribedBy
```

Comment:
now I have a doubt: maybe this is intentional? some sort of TS magic?

Reply:
yes, this decorator was introduced in #2177 and it's how we're now handling the descriptors in form components

Reply:
TIL